File: ch4/4-14.c
#include <stdio.h>

#define swap(t,x,y) { t temp; temp = (x); (x) = (y); (y) = temp; }

int main(void)
{
    int x = 3, y = 7;
    printf("x=%d, y=%d\n", x, y);
    swap(int, x, y);
    printf("x=%d, y=%d\n", x, y);
    return 0;
}

File: ch4/4-14.go
package main

import "fmt"

func main() {
	x, y := 3, 7
	fmt.Printf("x=%d, y=%d\n", x, y)
	swap(&x, &y)
	fmt.Printf("x=%d, y=%d\n", x, y)
}

func swap(x *int, y *int) {
	tmp := *x
	*x = *y
	*y = tmp
}
